File: packages/functions/src/sync-kb/lambda.py
import boto3
import os


def handler(event, context):
    print("Starting syncing...")
    print(os.environ['KB_NAME'])
    knowledge_base_id = event.get('knowledgeBaseId')
    data_source_id = event.get('dataSourceId')

    bedrock_client = boto3.client(
        service_name="bedrock-agent",
        region_name="us-east-1",
    )

    response = bedrock_client.start_ingestion_job(
        knowledgeBaseId=knowledge_base_id,
        dataSourceId=data_source_id,
    )

    return {
        "statusCode": 200,
        "headers": {
            "Content-Type": "application/json",
        },
        "body": {
            "ingestionJobId": response["ingestionJob"]["ingestionJobId"]
        },
    }

File: packages/functions/src/sync-kb/lambda.ts
import { APIGatewayProxyEvent, APIGatewayProxyResult, Context } from 'aws-lambda';
import { BedrockAgentClient, StartIngestionJobCommand } from '@aws-sdk/client-bedrock-agent';

const bedrockClient = new BedrockAgentClient({
  region: 'us-east-1',
});

export const handler = async (
  event: APIGatewayProxyEvent,
  context: Context
): Promise<APIGatewayProxyResult> => {
  console.log('Starting syncing...');
  
  try {
    const command = new StartIngestionJobCommand({
      knowledgeBaseId: process.env.KNOWLEDGE_BASE_ID!,
      dataSourceId: process.env.DATA_SOURCE_ID!,
    });

    const response = await bedrockClient.send(command);

    return {
      statusCode: 200,
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({
        ingestionJobId: response.ingestionJob?.ingestionJobId,
      }),
    };
  } catch (error) {
    console.error('Error:', error);
    return {
      statusCode: 500,
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({ error: 'Internal server error' }),
    };
  }
};